cfme/tests/infrastructure/test_instance_analysis.py

```diff
@@ -9,51 +9,100 @@
+from cfme.web_ui import InfoBlock, Table, SplitTable, paginator, DriftGrid
+from cfme.web_ui import tabstrip as tabs, toolbar as tb
 from fixtures.pytest_store import store
-from utils import testgen, ssh
+from utils import testgen, ssh, safe_string, version, error
 from utils.conf import cfme_data
+from utils.log import logger
 from utils.wait import wait_for
-from utils.version import pick, LOWEST
+
+WINDOWS = {'id': "Red Hat Enterprise Windows", 'icon': 'linux_windows'}
+
+RPM_BASED = {
+    'rhel': {
+        'id': "Red Hat", 'release-file': '/etc/redhat-release', 'icon': 'linux_redhat',
+        'package': "kernel", 'install-command': "",  # We don't install stuff on RHEL
+        'package-number': 'rpm -qa | wc -l'},
+    'centos': {
+        'id': "CentOS", 'release-file': '/etc/centos-release', 'icon': 'linux_centos',
+        'package': 'iso-codes', 'install-command': 'yum install -y {}',
+        'package-number': 'rpm -qa | wc -l'},
+    'fedora': {
+        'id': 'Fedora', 'release-file': '/etc/fedora-release', 'icon': 'linux_fedora',
+        'package': 'iso-codes', 'install-command': 'dnf install -y {}',
+        'package-number': 'rpm -qa | wc -l'},
+    'suse': {
+        'id': 'Suse', 'release-file': '/etc/SuSE-release', 'icon': 'linux_suse',
+        'package': 'iso-codes', 'install-command': 'zypper install -y {}',
+        'package-number': 'rpm -qa | wc -l'},
+}
+
+DEB_BASED = {
+    'ubuntu': {
+        'id': 'Ubuntu 14.04', 'release-file': '/etc/issue.net', 'icon': 'linux_ubuntu',
+        'package': 'iso-codes',
+        'install-command': 'env DEBIAN_FRONTEND=noninteractive apt-get -y install {}',
+        'package-number': "dpkg --get-selections | wc -l"},
+    'debian': {
+        'id': 'Debian ', 'release-file': '/etc/issue.net', 'icon': 'linux_debian',
+        'package': 'iso-codes',
+        'install-command': 'env DEBIAN_FRONTEND=noninteractive apt-get -y install {}',
+        'package-number': 'dpkg --get-selections | wc -l'},
+}
 
 
 def pytest_generate_tests(metafunc):
     # Filter out providers without templates defined
-    argnames, argvalues, idlist = testgen.infra_providers(metafunc, 'provisioning')
+    argnames, argvalues, idlist = testgen.infra_providers(metafunc, "vm_analysis_new")
 
-    new_argvalues = []
     new_idlist = []
+    new_argvalues = []
+
     for i, argvalue_tuple in enumerate(argvalues):
         args = dict(zip(argnames, argvalue_tuple))
-        if not args['provisioning']:
-            # Don't know what type of instance to provision, move on
-            continue
 
-        if not {'analysis-username', 'analysis-pass', 'analysis-image'}.issubset(
-                args['provisioning'].viewkeys()):
-            # Need all for template provisioning
+        vms = []
+        provisioning_data = []
+
+        try:
+            vms = argvalue_tuple[0].get("vms")
+            provisioning_data = argvalue_tuple[0].get("provisioning")
```

Comment:
We now use 'vm-analysis-new' key, which different structure, only 'image' and 'fs-type' are required